course3/homework-string/my_string.cpp

```diff
@@ -0,0 +1,209 @@
+#include "my_string.h"
+#include <memory>
+
+        MyString::MyString()
+        {
+            mBuffer=nullptr;
+        }
+
+        MyString::MyString(const char* other)
+        {
+            mBuffer = std::make_unique<char[]>(std::strlen(other) + 1);
+            std::strcpy(mBuffer.get(), other);
+        }
+
+        MyString::MyString(const MyString& other)
+        {
+            mBuffer = std::make_unique<char[]>(other.size() + 1);
+            std::strcpy(mBuffer.get(), other.mBuffer.get());
+        }
+
+        std::size_t MyString::size() const
+        {
+            if(!mBuffer)
+                return 0;
+
+            return std::strlen(mBuffer.get());
+        }
+
+        bool MyString::empty() const
+        {
+            return mBuffer == nullptr || mBuffer.get()[0] == 0;
+        }
+
+        const char* MyString::c_str() const
+        {
+            if(!mBuffer)
+                return "";
+
+            return mBuffer.get();
+        }
+
+        char MyString::operator[](std::size_t index) const
+        {
+            return mBuffer.get()[index];
+        }
+        char& MyString::operator[](std::size_t index)
+        {
+            return mBuffer.get()[index];
+        }
+
+        bool MyString::operator==(const MyString& other) const
+        {
+            if(this -> empty() && other.empty())
```

Comment:
Style: this time you use ```this->```, and in other methods you don't. Make your choice and stick to it.